gammapy/datasets/tests/test_simulate.py

```diff
@@ -126,22 +159,61 @@ def dataset():
 
 
 @requires_data()
-def test_sample_coord_time_energy(dataset, models):
-    models[0].spatial_model = None
-    dataset.models = models
+def test_evaluate_timevar_source(enedip_temporal_model, dataset):
+    dataset.models = enedip_temporal_model
+    evaluator = dataset.evaluators["test-source"]
+
+    sampler = MapDatasetEventSampler(random_state=0)
+    npred = sampler._evaluate_timevar_source(dataset, evaluator)
+
+    assert_allclose(np.shape(npred.data), (3, 1999, 1, 1))
+
+    assert_allclose(npred.data[:, 10, 0, 0], [0.024827, 0.110424, 0.321845], rtol=2e-4)
+    assert_allclose(npred.data[:, 50, 0, 0], [0.024827, 0.110424, 0.321845], rtol=2e-4)
+
+    filename = "$GAMMAPY_DATA/gravitational_waves/GW_example_DC_map_file.fits.gz"
+    temporal_model = LightCurveTemplateTemporalModel.read(filename, format="map")
+    temporal_model.t_ref.value = 51544.00074287037
+    dataset.models[0].temporal_model = temporal_model
+    evaluator = dataset.evaluators["test-source"]
+
+    sampler = MapDatasetEventSampler(random_state=0)
+    npred = sampler._evaluate_timevar_source(dataset, evaluator)
+
+    assert_allclose(
+        npred.data[:, 1000, 0, 0] / 1e-13,
+        [0.038113, 0.033879, 0.010938],
+        rtol=2e-4,
+    )
+
+
+@requires_data()
+def test_sample_coord_time_energy(dataset, enedip_temporal_model):
+    enedip_temporal_model.spatial_model = None
+    enedip_temporal_model.spectral_model = ConstantSpectralModel(
+        const="1 cm-2 s-1 TeV-1"
+    )
+    dataset.models = enedip_temporal_model
     evaluator = dataset.evaluators["test-source"]
     sampler = MapDatasetEventSampler(random_state=0)
     with pytest.raises(TypeError):
         sampler._sample_coord_time_energy(dataset, evaluator)
 
-    models[0].spatial_model = PointSpatialModel(
+    enedip_temporal_model.spatial_model = PointSpatialModel(
         lon_0="0 deg", lat_0="0 deg", frame="galactic"
     )
-    dataset.models = models
+    dataset.models = enedip_temporal_model
     evaluator = dataset.evaluators["test-source"]
     sampler = MapDatasetEventSampler(random_state=0)
-    with pytest.raises(NotImplementedError):
-        sampler._sample_coord_time_energy(dataset, evaluator)
+    events = sampler._sample_coord_time_energy(dataset, evaluator)
+
+    assert_allclose(len(events), 918)
+
+    assert_allclose(
+        [events[0][0], events[0][1], events[0][2], events[0][3]],
+        [568.238666, 7.687285, 266.404988, -28.936178],
```

Comment:
The test fail is on the simulated event energy?

Reply:
Yes, but locally it doesn't fail. So I don't understand why... any idea of what's going on?